app/uk/gov/hmrc/partnershipidentificationfrontend/service/AuditService.scala

```diff
@@ -113,4 +117,12 @@ class AuditService @Inject()(auditConnector: AuditConnector,
   private def identifiersMatchJson(identifiersMatch: Boolean): JsObject = Json.obj {
     "isMatch" -> identifiersMatch
   }
+
+  private def optionalJson(jsonKey: String, optValue: Option[String]): JsObject =
+    optValue match {
+      case Some(value) => Json.obj(
+        jsonKey -> value
+      )
+      case None => Json.obj()
+    }
 }
```

Comment:
While this reduced overall code, I feel that it is an unnecessary layer of abstraction. The pattern match within the original method is clear in what it is trying to achieve.

If we're going down this route, I believe there are better ways of doing this (https://github.com/hmrc/vat-registration/blob/main/app/utils/JsonUtils.scala). However, I don't feel that an audit service is necessarily complex to warrant this kind of approach.